cmd/start.go

```diff
@@ -698,9 +561,228 @@ func (x *Start) Execute(args []string) error {
+			addrutil.SupportedTransportStrings = append(addrutil.SupportedTransportStrings, "/onion")
+			t, err := ma.ProtocolsWithString("/onion")
+			if err != nil {
+				log.Error("wrapping onion protocol:", err)
+				return err
+			}
+			addrutil.SupportedTransportProtocols = append(addrutil.SupportedTransportProtocols, t)
+		} else {
+			usingClearnet = true
+		}
+	}
+	return nil
+}
+
+func getOnionAddress() (string, error) {
+	onionAddr, err := obnet.MaybeCreateHiddenServiceKey(repoPath)
+	if err != nil {
+		log.Error("create onion key:", err)
+		return "", err
+	}
+	return "/onion/" + onionAddr + ":4003", nil
+}
+
+// Configure IPFS Swarm for Tor and dual stack nodes
+func configureIPFSSwarmForTor(isTor bool, isDualStack bool) error {
+	onionAddrString, err := getOnionAddress()
+	if err != nil {
+		return err
+	}
+
+	if isTor {
+		ipfsConfig.Addresses.Swarm = []string{}
+		ipfsConfig.Addresses.Swarm = append(ipfsConfig.Addresses.Swarm, onionAddrString)
+	} else if isDualStack {
+		ipfsConfig.Addresses.Swarm = []string{}
+		ipfsConfig.Addresses.Swarm = append(ipfsConfig.Addresses.Swarm, onionAddrString)
+		ipfsConfig.Addresses.Swarm = append(ipfsConfig.Addresses.Swarm, "/ip4/0.0.0.0/tcp/4001")
+		ipfsConfig.Addresses.Swarm = append(ipfsConfig.Addresses.Swarm, "/ip6/::/tcp/4001")
+		ipfsConfig.Addresses.Swarm = append(ipfsConfig.Addresses.Swarm, "/ip6/::/tcp/9005/ws")
+		ipfsConfig.Addresses.Swarm = append(ipfsConfig.Addresses.Swarm, "/ip4/0.0.0.0/tcp/9005/ws")
+	}
+	return nil
+}
+
+// Retrieves identity key from the database and uses it to get the
+// IPFS identity which contains the private key and peer ID
+func getIPFSIdentity(db db.SQLiteDatastore) (config.Identity, error) {
+	identityKey, err := db.Config().GetIdentityKey()
+	if err != nil {
+		log.Error("get identity key:", err)
+		return config.Identity{}, err
+	}
+	identity, err := ipfs.IdentityFromKey(identityKey)
+	if err != nil {
+		log.Error("get identity from key:", err)
+	}
+	return identity, err
+}
+
+func getIPFSRepo() (ipfsrepo.Repo, error) {
+	repo, err := fsrepo.Open(repoPath)
+	if err != nil {
+		log.Error("open repo:", err)
+		return nil, err
+	}
+	return repo, nil
+}
+
+func getIPFSConfig() (*config.Config, error) {
+	var err error
+	nodeRepo, err = getIPFSRepo()
+	if err != nil {
+		return nil, err
+	}
+
+	cfg, err := nodeRepo.Config()
+	if err != nil {
+		log.Error("get repo config:", err)
+		return nil, err
+	}
+	return cfg, nil
+}
+
+// Configure the node's bootstrap peers and protocol tags for testnet
+func setupTestnet() error {
+	testnetBootstrapAddrs, err := schema.GetTestnetBootstrapAddrs(mainConfig)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	cfg, err := nodeRepo.Config()
+	cfg.Bootstrap = testnetBootstrapAddrs
+	dht.ProtocolDHT = "/openbazaar/kad/testnet/1.0.0"
+	bitswap.ProtocolBitswap = "/openbazaar/bitswap/testnet/1.1.0"
+	service.ProtocolOpenBazaar = "/openbazaar/app/testnet/1.0.0"
+
+	dataSharingConfig.PushTo = []string{}
+	return nil
+}
```

Comment:
Why not make a function which handles setting up peers on both mainnet and testnet?

Also, that this mutates the configs suggest they should be passed in as a param instead of grabbing from the global vars.